### PR TITLE
elsevier-package: propagate links in references to output

### PR DESCRIPTION
**Pull request by fschwenn**
* Propagates URLs which were already extracted now also to
  the final output.

Signed-off-by: fschwenn florian.schwennsen@desy.de

### CODE REVIEW

**Comment by kaplun** [comment]
@fschwenn seems like the test are not passing. Is there anything that you need to amend?

File: harvestingkit/elsevier_package.py [hunk 1]
```diff
@@ -361,6 +361,9 @@ def _add_references(self, xml_doc, rec, refextract_callback=None):
                         subfields.append(('s', journal))
                 if textref:
                     subfields.append(('m', textref))
+                if links:
```
Comment (tsgit) [inline harvestingkit/elsevier_package.py]: links defaults to `[]` so the if block isn't strictly necessary here

File: harvestingkit/elsevier_package.py [hunk 2]
```diff
@@ -295,7 +295,7 @@ def _normalize_article_dir_with_dtd(self, path):
     def _add_references(self, xml_doc, rec, refextract_callback=None):
         for label, authors, doi, issue, page, title, volume, year,\
```
Comment (tsgit) [inline harvestingkit/elsevier_package.py]: that's a lot of values to unpack by position and therefore error-prone and dangerous in a subtle way. using a data structure with labels like named tuples or dict would make this clearer and more robust

File: harvestingkit/elsevier_package.py [hunk 3]
```diff
@@ -764,7 +767,7 @@ def _get_ref(self, ref, label):
         year = re.sub(r'\D', '', year)
         return (label, authors, doi, issue, page, title, volume,
```
Comment (tsgit) [inline harvestingkit/elsevier_package.py]: see commentary above about long list of positional values

File: harvestingkit/elsevier_package.py (hunk 2 above)
Comment (fschwenn) [inline harvestingkit/elsevier_package.py]: I did not write the code, I'm just trying to fix a bug. I am not planning to rewrite the whole program with HEPCrawl arround the corner.

**Comment by kaplun** [comment]
@fschwenn but it seems to at least introduce one more bug (or maybe the tests need to be amended). Can you check:
https://travis-ci.org/inspirehep/harvesting-kit/builds/242372269#L569

**Comment by tsgit** [comment]
right, I pinpointed in my comments above why the test fails and it should be amended to take the added `links` member of a ref tuple into account

**Comment by fschwenn** [comment]
Thanks a lot, 
Florian 

-- 
Florian Schwennsen 
Deutsches Elektronen-Synchrotron DESY 
Building 01 Room O1.446 
phone: +49-40-8998-6190

**Pull request merged**